src/components/EventAdd/index.tsx

```diff
@@ -0,0 +1,180 @@
+import { DAYS } from '../../constants';
+import { ScheduleContext } from '../../contexts';
+
+import './stylesheet.scss';
+
+export type EventAddProps = {
+  className?: string;
+};
+
+export default function EventAdd({
+  className,
+}: EventAddProps): React.ReactElement {
+  const [{ events, colorMap }, { patchSchedule }] = useContext(ScheduleContext);
+  const [eventName, setEventName] = useState('');
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+  const [start, setStart] = useState<number>(-1);
+  const [end, setEnd] = useState<number>(-1);
+  const [submitDisabled, setSubmitDisabled] = useState(true);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    if (
+      eventName.length > 0 &&
+      selectedTags.length > 0 &&
+      start !== -1 &&
+      end !== -1 &&
+      !error
+    ) {
+      setSubmitDisabled(false);
+    } else {
+      setSubmitDisabled(true);
+    }
+  }, [eventName, selectedTags, start, end, error]);
+
+  function parseTime(time: string): number {
+    const split = time.split(':').map((str) => Number(str));
+
+    if (typeof split[0] !== 'undefined' && typeof split[1] !== 'undefined') {
+      return split[0] * 60 + split[1];
+    }
+    return -1; // invalid time string
+  }
+
+  function handleStartChange(event: React.ChangeEvent<HTMLInputElement>): void {
+    const newStart = parseTime(event.target.value);
+
+    setError('');
+    setStart(newStart);
+
+    if (end !== -1 && end <= newStart) {
+      setError('Start time must be before end time.');
+    }
+  }
+
+  function handleEndChange(event: React.ChangeEvent<HTMLInputElement>): void {
+    const newEnd = parseTime(event.target.value);
+
+    setError('');
+    setEnd(newEnd);
+
+    if (start !== -1 && newEnd <= start) {
+      setError('Start time must be before end time.');
+    }
+  }
+
+  const onSubmit = useCallback((): void => {
+    const id = new Date().getTime().toString();
+    const event = {
+      id,
+      name: eventName,
+      period: {
+        start,
+        end,
+      },
+      days: selectedTags,
+    };
+
+    patchSchedule({
+      events: [...castDraft(events), event],
+      colorMap: { ...colorMap, [id]: getRandomColor() },
+    });
+  }, [eventName, start, end, selectedTags, events, colorMap, patchSchedule]);
+
+  return (
+    <div className={classes('EventAdd', className)}>
+      <form className="add">
+        <table>
+          <tbody>
+            <tr>
+              <td>
+                <div
+                  className={classes('label', eventName.length > 0 && 'active')}
+                >
+                  Name
+                </div>
+              </td>
+              <td className="input">
+                <input
+                  type="text"
+                  value={eventName}
+                  onChange={(event): void => setEventName(event.target.value)}
+                  placeholder="Event Name"
+                />
```

Comment:
Add an `onKeyDown` option to allow users to submit when they press "Enter". Same thing for other input fields.